streams/src/error.rs

```diff
@@ -0,0 +1,122 @@
+//! Stream Errors
+
+// Rust
+use core::{array::TryFromSliceError, fmt::Debug};
+
+// 3rd-party
+use thiserror_no_std::Error;
+// IOTA
+
+// Streams
+use lets::{
+    address::{Address, MsgId},
+    error::Error as LetsError,
+    id::{Identifier, PskId},
+    message::{Topic, TopicHash},
+};
+
+use spongos::error::Error as SpongosError;
+
+pub type Result<T> = core::result::Result<T, Error>;
+
+#[derive(Debug, Error)]
+/// Error type of the iota client crate.
+#[allow(clippy::large_enum_variant)]
+/// #[serde(tag = "type", content = "error")]
+pub enum Error {
+    //////////
+    // Streams
+    //////////
+    #[error(
+        "address already taken for. The address '{1}' where the {0} message is being sent already contains some data, possibly spam."
+    )]
+    AddressUsed(&'static str, Address),
+
+    #[error("unexpected message type {0}")]
+    MessageTypeUnknown(u8),
+
+    #[error("message '{0}' not found in {1}")]
+    MessageMissing(MsgId, &'static str),
+
+    #[error("Failed getting messages for {0} due to {1}")]
```

Comment:
```suggestion
    #[error("Failed to get messages.  Error: {0}")]
```
I don't think we need to specify which function caused this error, it'll already be traced